src/NHibernate/Engine/Query/QueryPlanCache.cs

```diff
@@ -69,10 +69,14 @@ public IQueryExpressionPlan GetHQLQueryPlan(IQueryExpression queryExpression, bo
 
 				plan = new QueryExpressionPlan(queryExpression, shallow, enabledFilters, factory);
 				// 6.0 TODO: add "CanCachePlan { get; }" to IQueryExpression interface
-				if (!(queryExpression is ICacheableQueryExpression linqExpression) || linqExpression.CanCachePlan)
+				if(queryExpression is ICacheableQueryExpression linqExpression && linqExpression.CanCachePlan)
```

Comment:
Yeah, I've thought about this several times and also talked to a colleague about it. I really wasn't sure if this piece of code of you was intended or not. 

But honestly, my change wasn't meant to be committed. I simple forgot to revert it and want to apologize for wasting your time for this and other e.g. format changes.

Beside of that, this would be a clear change of the functionality of NHibernate.